pkg/ocmagenthandler/ocmagenthandler_pdb.go

```diff
@@ -0,0 +1,86 @@
+package ocmagenthandler
+
+import (
+	"context"
+	"reflect"
+
+	ocmagentv1alpha1 "github.com/openshift/ocm-agent-operator/api/v1alpha1"
+	oah "github.com/openshift/ocm-agent-operator/pkg/consts/ocmagenthandler"
+	v1 "k8s.io/api/policy/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+func buildOCMAgentPodDisruptionBudget(ocmAgent ocmagentv1alpha1.OcmAgent) *v1.PodDisruptionBudget {
+	namespacedName := oah.BuildNamespacedName(ocmAgent.Name + "-pdb")
+
+	return &v1.PodDisruptionBudget{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      namespacedName.Name,
+			Namespace: namespacedName.Namespace,
+		},
+		Spec: v1.PodDisruptionBudgetSpec{
+			MinAvailable: &intstr.IntOrString{
+				Type:   intstr.Int,
+				IntVal: 1,
+			},
```

Comment:
That means we will have to have at least 2 replicas, or else it will be non-disruptable 
Wondering if it would be better that we set this based on the ocmagent value? Not sure if it is needed.

Reply:
Thanks for your review, not sure if I understood your comment correctly.  I have create a PR to set the ocm-agent replicas to 2 in the [Operator Lifecycle Manager (OLM) template](https://github.com/openshift/ocm-agent/blob/e224d04693f988e5f3758d2c10ab3a137e6d6a2d/hack/00-ocm-agent.selectorsyncset.yaml.tmpl). Once it's merged, it can ensure that we have redundancy in our deployment and can provide HA.

With the `MinAvailable` set to 1 in the PDB, we ensure that at least one of these replicas is always available. This is aligned with our goal of maintaining high availability while allowing room for disruptions like maintenance or updates? 

Please correct me if I was wrong.

Reply:
yeah, I think it should be good.